charmstore/client.go

```diff
@@ -83,21 +82,111 @@ func newBaseClient(raw *csclient.Client, config ClientConfig, meta JujuMetadata)
 	base := &baseClient{
 		Client: raw,
 	}
-	base.asRepo = func() *charmrepo.CharmStore {
-		// TODO(ericsnow) Use charmrepo.NewCharmStoreFromClient(), when available?
-		repo := charmrepo.NewCharmStore(config.NewCharmStoreParams)
-		return repo.WithJujuAttrs(meta.asAttrs())
-	}
 	return base
 }
 
 // LatestRevisions implements BaseClient.
-func (base baseClient) LatestRevisions(cURLs []*charm.URL) ([]charmrepo.CharmRevision, error) {
+func (base baseClient) LatestRevisions(charms []CharmID) ([]charmrepo.CharmRevision, error) {
 	// TODO(ericsnow) Fix this:
 	// We must use charmrepo.CharmStore since csclient.Client does not
 	// have the "Latest" method.
-	repo := base.asRepo()
-	return repo.Latest(cURLs...)
+
+	// unfortunately, the way the csclient.Client uses channels is pretty
```

Comment:
s/unfortunately/Unfortunately/
Although perhaps just lose the word - it doesn't add much.